substrate/frame/support/src/transaction_extensions.rs

```diff
@@ -0,0 +1,155 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Transaction extensions.
+
+use crate::{CloneNoBound, EqNoBound, PartialEqNoBound, RuntimeDebugNoBound};
+use codec::{Codec, Decode, Encode};
+use scale_info::{StaticTypeInfo, TypeInfo};
+use sp_io::hashing::blake2_256;
+use sp_runtime::{
+	impl_tx_ext_default,
+	traits::{
+		transaction_extension::TransactionExtension, AsAuthorizedOrigin, DispatchInfoOf,
+		Dispatchable, IdentifyAccount, Verify,
+	},
+	transaction_validity::{InvalidTransaction, TransactionValidityError, ValidTransaction},
+};
+use sp_std::fmt::Debug;
+use sp_weights::Weight;
+
+/// Extension that, if enabled, validates a signature type against the payload constructed from the
+/// call and the rest of the transaction extension pipeline. This extension provides the
+/// functionality that traditionally signed transactions had with the implicit signature checking
+/// implemented in [`Checkable`](sp_runtime::traits::Checkable). It is meant to be placed ahead of
+/// any other extensions that do authorization work in the [`TransactionExtension`] pipeline.
```

Comment:
Documentation is not that great. I mean `UncheckedExtrinsic` for `Signed` still has a signature. So, this documentation first reads a little bit confusing.

Reply:
Also maybe giving an usage example.

Reply:
> I mean UncheckedExtrinsic for Signed still has a signature.

This extension is independent of `UncheckedExtrinsic`. The signed variant in `UncheckedExtrinsic` will be deprecated eventually.

This extension is used in the `pallet_example_authorization_tx_extension` example.